src/test/java/org/jenkinsci/plugins/workflow/steps/durable_task/ShellStepTest.java

```diff
@@ -128,6 +135,34 @@ public void failureShouldMarkNodeRed() throws Exception {
         assertTrue(found);
     }
 
+    /**
+     * Failure in the shell script should mark the step as red
+     */
+    @Test
+    public void stepDescription() throws Exception {
+        // job setup
+        WorkflowJob foo = j.jenkins.createProject(WorkflowJob.class, "foo");
+        foo.setDefinition(new CpsFlowDefinition(Functions.isWindows() ? "node {" +
+                "bat 'echo first';" +
+                "bat returnStdout: true, script: 'echo second';" +
+        "}" : "node {" +
+                "sh 'echo first'; " +
+                "sh returnStdout: true, script: 'echo second'" +
+        "}", true));
+
+        WorkflowRun b = j.buildAndAssertSuccess(foo);
+
+        ArrayList<String> args = new ArrayList<>();
+        List<FlowNode> shellStepNodes = new DepthFirstScanner().filteredNodes(b.getExecution(), new NodeStepTypePredicate(Functions.isWindows() ? "bat" : "sh"));
+        assertThat(shellStepNodes, Matchers.hasSize(2));
+        for(FlowNode n : shellStepNodes) {
+            args.add(ArgumentsAction.getStepArgumentsAsString(n));
+        }
+
+        assertArrayEquals(new String[] { "first", "second" }, args.toArray());
```

Comment:
BTW there is a more idiomatic way of doing this with `Matchers`, something like

```java
assertThat(args, contains("first", "second"));
```

but getting the `static import`s and generics exactly right can be tricky if you are not really comfortable with Java 5+.

Reply:
so @dwnusbaum provided a PR that fixed that too.